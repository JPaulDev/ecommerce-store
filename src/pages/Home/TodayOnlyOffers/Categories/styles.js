import styled from 'styled-components';

export const Container = styled.div`
  background-color: white;
  border: 1px solid var(--color-grey-9);
  margin-bottom: 15px;
  padding: 15px 5px;
`;

export const List = styled.ul`
  display: flex;
  justify-content: space-around;
  list-style: none;
  margin: 0 auto;
  max-width: 1100px;
`;
