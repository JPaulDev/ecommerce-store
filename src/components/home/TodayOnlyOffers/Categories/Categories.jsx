import {
  GraphicsCard,
  Memory,
  HardDrive,
  Processor,
  Motherboard,
} from '../../../icons';
import * as Styled from './styles';

const categories = [
  {
    heading: 'Graphics Cards',
    icon: <GraphicsCard width="60%" />,
  },
  {
    heading: 'Memory',
    icon: <Memory width="60%" />,
  },
  {
    heading: 'Storage',
    icon: <HardDrive width="60%" />,
  },
  {
    heading: 'Processors',
    icon: <Processor width="60%" />,
  },
  {
    heading: 'Motherboards',
    icon: <Motherboard width="60%" />,
  },
];

export default function Categories({ onChangeCategory }) {
  return (
    <Styled.Container>
      <Styled.List>
        {categories.map((item) => (
          <li key={item.heading}>
            <Styled.Button
              type="button"
              onClick={() => onChangeCategory(item.heading)}
            >
              <Styled.Wrapper>{item.icon}</Styled.Wrapper>
              <Styled.Heading>{item.heading}</Styled.Heading>
            </Styled.Button>
          </li>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
