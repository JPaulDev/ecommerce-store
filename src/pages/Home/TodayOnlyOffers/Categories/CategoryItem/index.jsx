import useIsHovered from '../../../../../hooks/useIsHovered';
import * as Styled from './styles';

export default function CategoryItem({ Icon, heading, onChangeCategory }) {
  const [isHovered, ref] = useIsHovered();

  return (
    <li ref={ref}>
      <Styled.Button type="button" onClick={onChangeCategory}>
        <Styled.Wrapper isHovered={isHovered}>
          <Icon />
        </Styled.Wrapper>
        <Styled.Heading isHovered={isHovered}>{heading}</Styled.Heading>
      </Styled.Button>
    </li>
  );
}
