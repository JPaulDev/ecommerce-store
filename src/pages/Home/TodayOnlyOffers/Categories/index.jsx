import { ReactComponent as GfxIcon } from '../../../../assets/svg/gfx.svg';
import { ReactComponent as MemoryIcon } from '../../../../assets/svg/memory.svg';
import { ReactComponent as HddIcon } from '../../../../assets/svg/hdd.svg';
import { ReactComponent as ProcessorIcon } from '../../../../assets/svg/processor.svg';
import { ReactComponent as MotherboardIcon } from '../../../../assets/svg/motherboard.svg';
import CategoryItem from './CategoryItem';
import * as Styled from './styles';

const categories = [
  {
    heading: 'Graphics Cards',
    icon: GfxIcon,
  },
  {
    heading: 'Memory',
    icon: MemoryIcon,
  },
  {
    heading: 'Hard Drives',
    icon: HddIcon,
  },
  {
    heading: 'Processors',
    icon: ProcessorIcon,
  },
  {
    heading: 'Motherboards',
    icon: MotherboardIcon,
  },
];

export default function Categories({ onChangeCategory }) {
  return (
    <Styled.Container>
      <Styled.List>
        {categories.map((category) => (
          <CategoryItem
            key={category.heading}
            Icon={category.icon}
            heading={category.heading}
            onChangeCategory={() => onChangeCategory(category.heading)}
          />
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
