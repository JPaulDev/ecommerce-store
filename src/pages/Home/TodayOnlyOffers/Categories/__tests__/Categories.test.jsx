import renderer from 'react-test-renderer';
import Categories from '../index';

describe('Categories component snapshot', () => {
  it('renders 5 buttons representing each selectable component category', () => {
    const tree = renderer.create(<Categories />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
