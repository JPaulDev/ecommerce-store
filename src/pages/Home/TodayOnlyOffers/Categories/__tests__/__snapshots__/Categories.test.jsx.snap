// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component snapshot renders 5 buttons representing each selectable component category 1`] = `
<div
  className="sc-hKMtZM ktgwFi"
>
  <ul
    className="sc-eCYdqJ fbmEHF"
  >
    <li>
      <button
        className="sc-bczRLJ dVIgNB"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gsnTZi iJClTJ"
        >
          <svg>
            gfx.svg
          </svg>
        </div>
        <h3
          className="sc-dkzDqf ildjWB"
        >
          Graphics Cards
        </h3>
      </button>
    </li>
    <li>
      <button
        className="sc-bczRLJ dVIgNB"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gsnTZi iJClTJ"
        >
          <svg>
            memory.svg
          </svg>
        </div>
        <h3
          className="sc-dkzDqf ildjWB"
        >
          Memory
        </h3>
      </button>
    </li>
    <li>
      <button
        className="sc-bczRLJ dVIgNB"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gsnTZi iJClTJ"
        >
          <svg>
            hdd.svg
          </svg>
        </div>
        <h3
          className="sc-dkzDqf ildjWB"
        >
          Storage
        </h3>
      </button>
    </li>
    <li>
      <button
        className="sc-bczRLJ dVIgNB"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gsnTZi iJClTJ"
        >
          <svg>
            processor.svg
          </svg>
        </div>
        <h3
          className="sc-dkzDqf ildjWB"
        >
          Processors
        </h3>
      </button>
    </li>
    <li>
      <button
        className="sc-bczRLJ dVIgNB"
        onClick={[Function]}
        type="button"
      >
        <div
          className="sc-gsnTZi iJClTJ"
        >
          <svg>
            motherboard.svg
          </svg>
        </div>
        <h3
          className="sc-dkzDqf ildjWB"
        >
          Motherboards
        </h3>
      </button>
    </li>
  </ul>
</div>
`;
